app/controllers/fhir/r4/api_controller.rb

```diff
@@ -396,10 +396,11 @@ def search
     summary_mode = page_size.zero?
     page = params.permit(:page)[:page].to_i
     page = 1 if page.zero?
-    results = []
-    unless summary_mode || resources.blank?
```

Comment:
FYI, calling `resources.blank?` was the culprit behind a query for all patients. Switching it to `!resources.exists` makes it much faster, but even that is unnecessary